src/vm.rs

```diff
@@ -748,9 +747,9 @@ impl<'a, S: Storage> Vm<'a, S> {
     }
 }
 
-pub(crate) fn build_evm<'a, DB: Database>(
+pub(crate) fn build_evm<'a, DB: Database, C: PevmChain>(
     db: DB,
-    chain: Chain,
+    chain: &C,
```

Comment:
Looking back, propagating a lot of PevmChain here seems overkill. We can just pass in a chain_id instead?

Reply:
as discussed, we need `PevmChain` in `build_evm`